proposals/VK_EXT_map_memory_placed.asciidoc

```diff
@@ -0,0 +1,145 @@
+
+The following features are exposed by this extension:
+
+[source,c]
+----
+typedef enum VkMemoryMapFlagBits {
+    VK_MEMORY_MAP_PLACED_BIT_EXT = 0x00000001,
+} VkMemoryMapFlagBits;
+
+typedef enum VkMemoryUnmapFlagBits {
+    VK_MEMORY_UNMAP_RESERVE_BIT_EXT = 0x00000001,
+} VkMemoryMapFlagBits;
+----
+
+ . `memoryMapPlaced` indicates support for placed memory maps
+ . `memoryMapRangePlaced` indicates support for placed memory maps of
+   subranges of the `VkDeviceMemory` object.
+ . `memoryUnmapReserve` indicates support for
+   `VK_MEMORY_UNMAP_RESERVE_BIT_EXT`.
+
+When `VK_MEMORY_MAP_PLACED_BIT_EXT` is set in the `flags` member of
+`VkMemoryMapInfoEXT` and a `VkMapMemoryPlacedInfoEXT` structure is included
+in the `pNext` chain of `VkMemoryMapInfoEXT`, the implementation attempts
+to place the map at the address specified by
+`VkMapMemoryPlacedInfoEXT::pPlacedAddress`.
+If the map cannot be placed exactly at the specified virtual address,
+`vkMapMemory2EXT()` returns `VK_ERROR_MEMORY_MAP_FAILED`.
+
+When `VK_MEMORY_UNMAP_RESERVE_BIT_EXT` is set in the `flags` member of
+`VkMemoryUnmapInfoEXT`, the address range currently occupied by the memory
+map will be atomically re-reserved by `vkUnmapMemory()` so that another
+call to `mmap()` or `vkMapMemory()` will not re-use that address range
+without an address hint or `VK_MEMORY_MAP_PLACED_BIT_EXT`, respectively.
+
+## Examples
+
+The following example allocates a memory object and maps it in the lower 32
+bit address space, using `mmap()` to select the address.
+
+[source,c]
+----
+VkDeviceMemory mem;
+const VkMemoryAllocateInfo allocInfo = {
+    .sType = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO,
+   .allocationSize = /* compute size */,
+   .memoryTypeIndex = /* compute memory type */,
+};
+VK_CHECK(vkAllocateMemory(device, &allocInfo, NULL, &mem));
+
+void *reserved = mmap(NULL, allocInfo.allocationSize, PROT_NONE,
+                      MAP_ANONYMOUS | MAP_PRIVATE | MAP_32BIT, -1, 0);
+CHECK(reserved != MAP_FAILED);
+
+const VkMemoryMapPlacedInfoEXT mapPlacedInfo = {
+    .sType = VK_STRUCTURE_TYPE_MEMORY_MAP_PLACED_INFO_EXT,
+    .pPlacementAddress = reserved,
+};
+const VkMemoryMapInfoEXT mapInfo = {
+    .sType = VK_STRUCTURE_TYPE_MEMORY_MAP_INFO_EXT,
+    .pNext = &mapPlacedInfo,
+    .memory = mem,
+    .offset = 0,
+    .size = VK_WHOLE_SIZE,
+    .flags = VK_MEMORY_MAP_PLACED_BIT_EXT,
+};
+void *map;
+VK_CHECK(vkMapMemory2EXT(device, &mapInfo, &map));
+CHECK(map == reserved);
+----
+
+## Issues
+
+1) How should the requested address be passed to `vkMapMemory()`?
+
+*PROPOSED*: When `VK_MEMORY_MAP_PLACED_BIT_EXT` is passed to
+`vkMapMemory()`, the `ppData` parameter becomes an input and the
+implementation attempts to map to whatever address is pointed to by
+`ppData`.
+
+2) How can a client atomically re-reserve the address range on unmap?
+
+*PROPOSED*: When `VK_MEMORY_MAP_RESERVE_ON_UNMAP_BIT_EXT` is passed to
+`vkMapMemory()`, the implementation remembers this bit and, when
```

Comment:
On line 59 a VK_MEMORY_UNMAP_RESERVE_BIT_EXT flag is mentioned for vkUnmapMemory2EXT, which contradicts this statement I think.

Reply:
Fixed.  I found a couple other simliar issues which I've now fixed up as well.